mantle/cmd/ore/aws/upload.go

```diff
@@ -140,8 +139,21 @@ func runUpload(cmd *cobra.Command, args []string) error {
 		fmt.Fprintf(os.Stderr, "Unrecognized args in aws upload cmd: %v\n", args)
 		os.Exit(2)
 	}
-	if uploadDiskSizeInspect && (uploadSourceObject != "" || uploadSourceSnapshot != "") {
-		fmt.Fprintf(os.Stderr, "--disk-size-inspect cannot be used with --source-object or --source-snapshot.\n")
+	if uploadSourceObject != "" && uploadSourceSnapshot != "" {
+		fmt.Fprintf(os.Stderr, "At most one of --source-object and --source-snapshot may be specified.\n")
+		os.Exit(2)
+	}
+	// The Windows License Included (winli) image is built using a source snapshot, but the disk size is not known.
+	// Allow setting uploadDIskSIzeInspect when using a source snapshot to discover the size of the disk from the
+	// aws upload file that was used to create the snapshot.
```

Comment:
hmm. I'm not the biggest fan of this change.

Why not just query to get the size of the source snapshot?

Reply:
i.e. we can pass a blank value for `uploadDiskSizeGiB` to `API.CreateHVMImage` if we're providing `sourceSnapshot` and have `CreateHVMImage` figure it out (if it's even required in the first place).

Reply:
`uploadDiskSizeGiB` is [never blank](https://github.com/coreos/coreos-assembler/blob/eeafb934dfffbeb003e3e46d3fa0e31c80784cb2/mantle/cmd/ore/aws/upload.go#L82). I suppose we could change the [default](https://github.com/coreos/coreos-assembler/blob/eeafb934dfffbeb003e3e46d3fa0e31c80784cb2/mantle/platform/api/aws/images.go#L37) to `16`, or we could query the source snapshot.